numpy/lib/stride_tricks.py

```diff
@@ -111,6 +112,164 @@ def as_strided(x, shape=None, strides=None, subok=False, writeable=True):
     return view
 
 
+def _sliding_window_view_dispatcher(x, shape, axis=None, *,
+                                    subok=None, writeable=None):
+    return (x,)
+
+
+@array_function_dispatch(_sliding_window_view_dispatcher, module='numpy')
+def sliding_window_view(x, shape, axis=None, *, subok=False, writeable=False):
+    """
+    Create a sliding window view into the array with the given shape.
+
+    Creates a sliding window view of the N dimensional array with the given
+    window shape. Window slides across each dimension of the array and extract
+    a subsets of the array at any window position.
+
+    Parameters
+    ----------
+    x : array_like
+        Array to create the sliding window view from.
+
+    shape : int or tuple of int
+        The shape of the window. If `axis` is not present, must have same
+        length as the number of input array dimensions.
+        Single integers `i` are treated as if they were the tuple `(i,)`.
+
+    axis : int or tuple of int, optional
+        If present, `shape[i]` will refer to the axis `axis[i]` of `x`,
+        otherwise `shape[i]` will refer to axis `i` of `x`.
```

Comment:
I wonder if it makes more sense for the default to be "all axes". Has this been considered?

Reply:
I'm not sure I completely understand what you mean. In a sense, the default already is "all axes" in so far as `axis=None` or not present is equivalent to `axis=tuple(range(x.ndim))` and means you need to specify as many entries in `shape`.

Perhaps your suggestion is to change the meaning of single integers in `shape` from `(i,)` to `(i,)*len(axis)`?

Reply:
Another option that @shoyer might be suggesting that `axis=-1` or `axis=0` is a sensible default. This might be the type of thing where the best thing to do is describe the possible choices at the top of this issue, and ultimately let the mailing list decide.

Reply:
I misinterpreted the docstring. I was trying to suggest exactly what is already implemented!

I would suggest revising this to something like:

> Axis or axes along which the sliding window is applied. By default, the sliding window is applied to all axes.

And then for window_shape:

> Size of window over each axis that takes part in the sliding window...

Reply:
Ok, seems the implementation is agreed; will work on the docstring.

Reply:
Clarified docstring.